### PR TITLE
IBX-2362: Added dynamic targeting block

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-2362
| Versions      | master, 4.1, 4.0

### CODE REVIEW

**Review by ciastektk — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Review by dkorostynski — APPROVED** [review]

File: docs/site_organization/working_with_page.md [hunk 1]
```diff
@@ -78,7 +78,8 @@ The blocks provided with a clean installation of [[= product_name_ee =]] are:
 
 |Block|Description|
 |-----|-----------|
-|Targeting|Embeds a Content item depending on the [Segment](organizing_the_content.md#segments) the user belongs to. For details, see [Targeting block](#targeting-block).|
+|Targeting|Embeds an Content item depending on the [Segment](organizing_the_content.md#segments) the user belongs to. For details, see [Targeting block](#targeting-block).|
+|Dynamic targeting|Embeds recommended items depending on the [Segment](organizing_the_content.md#segments) the user belongs to. For details, see [Dynamic targeting](#dynamic-targeting).|
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:82]:
```suggestion
|Targeting|Embeds Content items based on the [Segment](organizing_the_content.md#segments) the user belongs to. For details, see [Targeting block](#targeting-block).|
|Dynamic Targeting|Embeds recommended items based on the [Segment](organizing_the_content.md#segments) the user belongs to. For details, see [Dynamic Targeting block](#dynamic-targeting-block).|
```

File: docs/site_organization/working_with_page.md [hunk 2]
```diff
@@ -116,6 +117,8 @@ After you change the block settings, click **Submit** to save the changes or **D
 
 ### Targeting block
 
+Dynamic targeting block provides recommendation of content based on users related to the configured Segments.
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:120]:
```suggestion
Targeting block provides Content item recommendations based on user groups that are related to selected Segments.
```

File: docs/site_organization/working_with_page.md [hunk 3]
```diff
@@ -130,6 +133,21 @@ You can preview the page for each of the available Segments:
 
 ![Previewing Page for a given Segment](img/page_builder_segment_preview.png)
 
+### Dynamic targeting
+
+Dynamic targeting block provides recommendation items based on users related to the configured Segments.
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:138]:
```suggestion
Dynamic targeting block provides recommended items based on scenarios that apply to users related to the configured Segments.
```

File: docs/site_organization/working_with_page.md [hunk 4]
```diff
@@ -130,6 +133,21 @@ You can preview the page for each of the available Segments:
...
+
+On the **Basic** tab, perform the following actions:
+
+- In the **Name** field, provide the name of the block.
+- In the **Select default scenario** section, select the default scenario for recommendation items that should be rendered if the current user
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:143]:
```suggestion
- In the **Select default scenario** section, select the default scenario to generate a list of recommended items that should be rendered 
```

File: docs/site_organization/working_with_page.md [hunk 5]
```diff
@@ -130,6 +133,21 @@ You can preview the page for each of the available Segments:
...
+is not assigned to any Segments.
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:144]:
```suggestion
if the current user is not assigned to any Segment.
```

File: docs/site_organization/working_with_page.md [hunk 6]
```diff
@@ -130,6 +133,21 @@ You can preview the page for each of the available Segments:
...
+- In the **Setup segment and scenario matching priority rules** drop-down, select a Segment group, a Segment identifier and Scenario that you want to display recommendations from.
+
+The rules are checked in order, so if a user belongs to more than one Segment, the first rule applies.
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:147]:
```suggestion
The rules are checked in order, so when a user belongs to more than one Segment, the first rule applies.
```

**Review by dabrt — APPROVED** [review]

File: docs/site_organization/working_with_page.md [hunk 7]
```diff
@@ -130,6 +133,21 @@ You can preview the page for each of the available Segments:
 
 ![Previewing Page for a given Segment](img/page_builder_segment_preview.png)
 
+### Dynamic targeting
```
[the remark this answers is not in the document]
Comment (juskora) [reply docs/site_organization/working_with_page.md:136]: The label is as on the UI: Dynamic targeting

**Pull request merged**